iroh-net/src/dns.rs

```diff
@@ -97,10 +100,69 @@ pub trait ResolverExt {
         host: N,
         timeout: Duration,
     ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
+
+    /// Looks up node info by DNS name.
+    fn lookup_by_domain(&self, name: &str) -> impl Future<Output = Result<NodeAddr>>;
+
+    /// Looks up node info by [`NodeId`] and origin domain name.
+    fn lookup_by_id(
+        &self,
+        node_id: &NodeId,
+        origin: &str,
+    ) -> impl Future<Output = Result<NodeAddr>>;
+
+    /// Perform an ipv4 lookup with a timeout in a staggered fashion.
+    ///
+    /// The first call is done immediately, with added calls according to the staggering strategy.
+    fn staggered_lookup_ipv4<N: IntoName + Clone>(
+        &self,
+        host: N,
+        timeout: Duration,
+        delays_ms: &[u64],
+    ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
```

Comment:
we can use `async fn` in traits now I think :=

Reply:
this generates a warning, let me paste it

Reply:
```
warning: use of `async fn` in public traits is discouraged as auto trait bounds cannot be specified
  --> iroh-net/src/dns.rs:84:5
   |
84 |     async fn lookup_ipv4<N: IntoName>(
   |     ^^^^^
   |
   = note: you can suppress this lint if you plan to use the trait only in your own code, or do not care about auto traits like `Send` on the `Future`
```

Reply:
ah okay, I didn't realize this was the case

Reply:
the implementation itself does use async, the declaration doesn't, unless we suppress the warning . What do you think?